src/rgw/rgw_op.cc

```diff
@@ -901,6 +901,12 @@ void rgw_build_iam_environment(rgw::sal::Store* store,
     s->env.emplace("aws:UserAgent", i->second);
   }
 
+  i = m.find("QUERY_STRING"); {
+    std::size_t pos = (i->second).find("versionId");
+    if( pos != string::npos)
+      s->env.emplace("s3:VersionId", (i->second).substr(pos+10,31));
```

Comment:
What will be the actual string here and why does it need to be trimmed?

Reply:
["QUERY_STRING"] = "versionId=zWsV8BClqWUUosc-CLysDwhxV2Pufub"  , but I observed that this may vary for other cases, our intention is fetch version id alone

Reply:
Please confirm if version id can have '='. 
If not, please split the string by '=' and use the later part.

Reply:
@shriya-deshmukh can you please confirm.

Reply:
**'=', '/', '+'** these characters are not present in the version-id.

Reply:
@shriya-deshmukh can you please confirm "&" is part of version id?

Reply:
No '&' will not be part of version-id.

Possible characters present in version-id - 
`ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-.`